### ISSUE TITLE
Authorization not working with Karma Process

### ISSUE THREAD

**Issue by dheerajkr126** [prymitive/karma#2140]
Hello Team,

I am trying to implement authentication and authorization for silencing Alertmanager Alerts. 
After searching on the internet found that Karma Tools is best for it. 
But when I have implemented with karma into Linux environment only authentication working and seems authorization role is bypassing by the karma process. 

./karma-linux-amd64 --version
>> v0.68

alertmanager, version 0.21.0

acls.yaml
------
rules:
  - action: block
    reason: all regex silences are blocked, use only concrete label names and values
    scope:
      filters:
        - name_re: .+
          value_re: .+
          isRegex: true
  - action: block
    reason: not allowed cluster=prod
    scope:
      filters:
        - name: cluster
          value: prod
  - action: requireMatcher
    reason: prometheus admins must add techteam=prometheus to all silences
    scope:
      groups:
        - prometheusAdmins
    matchers:
      required:
        - name: techteam
          value: prometheus
----- 

karma.yaml
------
authentication:
  basicAuth:
    users:
      - username: user1
        password: xxxxxxx
      - username: user2
        password: xxxxxxx

authorization:
  acls:
    silences: /<path>/acls.yaml
  groups:
    - name: prometheusAdmins
      members:
        - user1

alertmanager:
  interval: 60s
  servers:
    - name: dev-alertmanager
      uri: url_var
      timeout: 10s
      proxy: true
      readonly: false
------ 

We are expecting here but all the below rules are bypassed- 

* No users should pass regex into silence api || Bypassed
* No One makes silence with cluster=prod label. || Bypassed
* prometheusAdmins should have label techteam=prometheus into silencing api  || Bypassed

**Comment by prymitive** [comment]
# Setup 
I edited [demo/acls.yaml](https://github.com/prymitive/karma/blob/master/demo/acls.yaml) to have this content:

```YAML
rules:
  - action: block
    reason: all regex silences are blocked, use only concrete label names and values
    scope:
      filters:
        - name_re: .+
          value_re: .+
          isRegex: true
  - action: block
    reason: not allowed cluster=prod
    scope:
      filters:
        - name: cluster
          value: prod
  - action: requireMatcher
    reason: prometheus admins must add techteam=prometheus to all silences
    scope:
      groups:
        - prometheusAdmins
    matchers:
      required:
        - name: techteam
          value: prometheus
```

Then I edited top of [demo/karma.yaml](https://github.com/prymitive/karma/blob/master/demo/karma.yaml) to have:

```YAML
authentication:
  basicAuth:
    users:
      - username: user1
        password: user1
      - username: user2
        password: user2
authorization:
  acl:
    silences: /etc/acls.yaml
  groups:
    - name: prometheusAdmins
      members:
        - user1
alertmanager:
  interval: 10s
[...]
```

I started demo docker:

```
% make run-demo
docker build --build-arg VERSION=v0.68-254-g5cd27b18-dev -t karma:demo -f demo/Dockerfile .
Sending build context to Docker daemon  147.7MB
Step 1/38 : FROM node:12.18.3-alpine as nodejs-builder
 ---> 18f4bc975732
Step 2/38 : RUN mkdir -p /src/ui
 ---> Using cache
 ---> d1ae4848b97f
Step 3/38 : COPY ui/package.json ui/package-lock.json /src/ui/
[...]
docker run --rm --name karma-demo -p 8080:8080 -p 9093:9093 -p 9094:9094 -p 9095:9095 karma:demo
2020-09-03 16:59:49,825 INFO supervisord started with pid 1
2020-09-03 16:59:50,834 INFO spawned: 'alertmanager1' with pid 8
2020-09-03 16:59:50,838 INFO spawned: 'alertmanager2' with pid 9
2020-09-03 16:59:50,842 INFO spawned: 'alertmanager3' with pid 14
2020-09-03 16:59:50,848 INFO spawned: 'generator' with pid 15
2020-09-03 16:59:50,866 INFO spawned: 'karma' with pid 23
2020-09-03 16:59:50,869 INFO spawned: 'kthxbye' with pid 25
```

And tried to create silences:

# No users should pass regex into silence api 

Created a silence with a regex:

<img width="835" alt="Screenshot 2020-09-03 at 18 00 35" src="https://user-images.githubusercontent.com/114288/92145372-08a2dd00-ee10-11ea-9e06-2f8ffa9146a1.png">

Then trying to submit it:
<img width="825" alt="Screenshot 2020-09-03 at 18 00 42" src="https://user-images.githubusercontent.com/114288/92145382-0c366400-ee10-11ea-8c5e-ce952d42d29b.png">

# No One makes silence with cluster=prod label

<img width="845" alt="Screenshot 2020-09-03 at 18 00 51" src="https://user-images.githubusercontent.com/114288/92145421-21ab8e00-ee10-11ea-98f5-7b490d5d8755.png">

<img width="821" alt="Screenshot 2020-09-03 at 18 00 58" src="https://user-images.githubusercontent.com/114288/92145429-24a67e80-ee10-11ea-97d3-ef45938a45b7.png">

# prometheusAdmins should have label techteam=prometheus into silencing api

<img width="826" alt="Screenshot 2020-09-03 at 18 01 20" src="https://user-images.githubusercontent.com/114288/92145464-2ec87d00-ee10-11ea-8830-3177a28d29ee.png">

<img width="835" alt="Screenshot 2020-09-03 at 18 01 27" src="https://user-images.githubusercontent.com/114288/92145492-3425c780-ee10-11ea-98a5-e5d8285641dc.png">

# tl;dr

Can't reproduce, how are you testing it?

**Comment by dheerajkr126** [comment]
Thanks, @prymitive for looking at this issue very quickly. 

There was yaml field issue into karma.yaml config file it should be **acl** not **acls** field inside authorization YAML fields. 
I have followed this document for acl creation where acls field is mention not acl under authorization yaml fields. 

https://github.com/prymitive/karma/blob/master/docs/CONFIGURATION.md
--------- 
authentication:
  basicAuth:
    users:
      - username: alice
        password: secret
      - username: bob
        password: secret
      - username: john
        password: secret
authorization:
  **acls**:
    silences: /etc/karma/acls.yaml
  groups:
    - name: admins
      members:
        - alice
        - bob
    - name: users
      members:
        - john
------ 

I think documentation is not updated so this type of issue we faced. Now after changing from acls --> acl , ACLS rules are working fine. Thanks for looking out. 

karma.yaml
----
authentication:
basicAuth:
users:
- username: user1
password: xxxxxxx
- username: user2
password: xxxxxxx

authorization:
**acl:**
silences: //acls.yaml
groups:
- name: prometheusAdmins
members:
- user1
----- 
[https://github.com/prymitive/karma/blob/master/docs/CONFIGURATION.md](url)
